### PR TITLE
feature: android SDK sample code

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* feature: android SDK sample code

## Linear Issue Reference

* [Android SDK Sample Code (LOG-263)](https://linear.app/silverhand/issue/LOG-263/android-sdk-sample-code)

## Todo
* [organize sdk library & sample code library (LOG-317)](https://linear.app/silverhand/issue/LOG-317/organize-sdk-library-and-sample-code-library)

## Testing
https://user-images.githubusercontent.com/10806653/142348827-ea17ef63-4c30-4fcc-965c-5544364caa77.mp4

### CODE REVIEW

**Review by wangsijie — CHANGES_REQUESTED** [review]
should we provider examples of "recover tokenSet from local cache" and "go into a page showing claims when is authenticated"

**Review by wangsijie — CHANGES_REQUESTED** [review]
should we provider examples of "recover tokenSet from local cache" and "go into a page showing claims when is authenticated"

File: android/app/src/main/java/io/logto/demo/MainActivity.kt [hunk 1]
```diff
@@ -1,11 +1,96 @@
...
+            refreshAccessToken()
+        }
+    }
+
+    private fun signIn() {
+        logto.signInWithBrowser(this) { exception, tokenSet ->
+            if (exception !== null) {
+                Log.d(TAG, "Sign in Failed: $exception")
+                return@signInWithBrowser
+            } else {
```
Comment (wangsijie) [inline android/app/src/main/java/io/logto/demo/MainActivity.kt:55]: useless `else`

File: android/app/src/main/java/io/logto/demo/MainActivity.kt [hunk 2]
```diff
@@ -1,11 +1,96 @@
...
+            }
+        }
+    }
+
+    private fun signOut() {
+        logto.signOutWithBrowser(this) { exception ->
+            if (exception != null) {
+                Log.d(TAG, "Sign out with exception: $exception")
+                return@signOutWithBrowser
+            } else {
```
Comment (simeng-li) [inline android/app/src/main/java/io/logto/demo/MainActivity.kt:67]: ditto

File: android/app/src/main/java/io/logto/demo/viewmodel/LogtoViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        }
+    }
+
+    fun signOut(context: Context) {
+        logto.signOutWithBrowser(context) { exception ->
+            if (exception != null) {
+                _logtoException.postValue(exception)
+            }
+            _authenticated.postValue(logto.authenticated)
```
Comment (simeng-li) [inline android/app/src/main/java/io/logto/demo/viewmodel/LogtoViewModel.kt:49]: should logout ahead

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: android/app/src/main/java/io/logto/demo/viewmodel/LogtoViewModel.kt (hunk 3 above)
Comment (xiaoyijun) [reply android/app/src/main/java/io/logto/demo/viewmodel/LogtoViewModel.kt:49]: Todo: [Align sign out process with js-sdk (LOG-265)](https://linear.app/silverhand/issue/LOG-265/align-sign-out-process-with-js-sdk)

**Review by wangsijie — APPROVED** [review]

**Pull request merged**